jans-cedarling/cedarling/src/jwt/token.rs

```diff
@@ -0,0 +1,227 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use crate::common::policy_store::{ClaimMappings, TokenEntityMetadata, TokenKind, TrustedIssuer};
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+
+const DEFAULT_USER_ID_SRC_CLAIM: &str = "sub";
+const DEFAULT_ROLE_SRC_CLAIM: &str = "role";
+
+pub enum TokenStr<'a> {
+    Access(&'a str),
+    Id(&'a str),
+    Userinfo(&'a str),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Token<'a> {
+    pub kind: TokenKind,
+    pub iss: Option<&'a TrustedIssuer>,
+    claims: TokenClaims,
+}
+
+impl<'a> Token<'a> {
+    pub fn new_access(claims: TokenClaims, iss: Option<&'a TrustedIssuer>) -> Token<'a> {
+        Self {
+            kind: TokenKind::Access,
+            iss,
+            claims,
+        }
+    }
+
+    pub fn new_id(claims: TokenClaims, iss: Option<&'a TrustedIssuer>) -> Token<'a> {
+        Self {
+            kind: TokenKind::Id,
+            iss,
+            claims,
+        }
+    }
+
+    pub fn new_userinfo(claims: TokenClaims, iss: Option<&'a TrustedIssuer>) -> Token<'a> {
+        Self {
+            kind: TokenKind::Userinfo,
+            iss,
+            claims,
+        }
+    }
+
+    #[allow(dead_code)]
```

Comment:
Why we need this?

Reply:
we have a `TokenKind::Transaction` so i added it for completeness' sake.

Reply:
i just removed it here since it seems easy enough to add back later: [87b399f](https://github.com/JanssenProject/jans/pull/10436/commits/87b399ff0bfa27c878c407a1121fb9082d2cafce).